fairscale/experimental/wgit/signal_sparsity.py

```diff
@@ -52,28 +108,81 @@ class SignalSparsity:
+        sst_top_k_percent: Optional[float] = None,
+        dst_top_k_dim: Optional[int] = -1,
+        dst_top_k_element: Optional[int] = None,
+        dst_top_k_percent: Optional[float] = None,
+    ) -> None:
+
+        self._sst_top_k_dim = sst_top_k_dim
+        self._sst_top_k_element = sst_top_k_element
+        self._sst_top_k_percent = sst_top_k_percent
+        self._dst_top_k_dim = dst_top_k_dim
+        self._dst_top_k_element = dst_top_k_element
+        self._dst_top_k_percent = dst_top_k_percent
+
         self._validate_conf()
+        self._transform = torch.fft.fft if algo == Algo.FFT else self._dct_transform  # type: ignore
 
     def _validate_conf(self) -> None:
         """Validating the config is valid.
 
-        For example, not both top_k_element and top_k_percent is set.
+        This includes asserting the following:
+        1. validating that both top_k_element and top_k_percent is not set.
+        2. Asserting that if top_k_percent is not None, it is a valid number for percentage.
 
-        this should assert fail if checking fails.
+        This asserts fail if the checking fails.
         """
-        pass
+        # assert that both top_k_elements and top_k_percent aren't set for sst and dst
+        assert (self._sst_top_k_element is None) ^ (
+            self._sst_top_k_percent is None
+        ), "One and only one of sst_top_k_element and sst_top_k_percent has to be provided as argument"
+        assert (self._dst_top_k_element is None) ^ (
+            self._dst_top_k_percent is None
+        ), "One and only one of dst_top_k_element and dst_top_k_percent has to be provided as argument"
+
+        # assert, if top_k_percent is not None, it is a valid number for percentage.
+        assert self._sst_top_k_percent is None or (
+            0.0 < self._sst_top_k_percent <= 100.0
+        ), "The value for the argument sst_top_k_percent has to be in the interval (0, 100]"
+
+        assert self._dst_top_k_percent is None or (
+            0.0 < self._dst_top_k_percent <= 100.0
+        ), "The value for the argument dst_top_k_percent has to be in the interval (0, 100]"
 
     def dense_to_sst(self, dense: Tensor) -> Tensor:
         """Get SST from a tensor
 
         Dense -> fft -> top-k -> results.
 
+        Generates a Signal Sparse Tensor (SST) from an input dense tensor. The input dense
+        tensor is transformed using a transform algorithm according to the `algo` initialization
+        argument. The SST is then generated from the top_k_elements (or the top_k_percentage) of
+        values from the transformed tensor along the 'sst_top_k_dim'.
+
+        Args:
+            dense (Tensor):
+                Input dense tensor (no zeros).
+
         Returns:
             Same shaped tensor, still in dense format but in frequency domain and has zeros.
         """
-        pass
+        k = get_k_for_topk(dense, self._sst_top_k_percent, self._sst_top_k_element, self._sst_top_k_dim)
```

Comment:
we can reduce the helper function's input to just k_ele, k_percent and top_k_total_size? That way, you don't need to pass both tensor and dim into it.
Have a different function that goes from (tensor, dim) -> top_k_total_size could be a reusable thing.